In [ ]:
import pandas as pd
import numpy as np
from scipy.stats import norm

In [ ]:
from google.colab import files
uploaded = files.upload()
df = pd.read_csv("TeachingRatings.csv")

Saving TeachingRatings.csv to TeachingRatings.csv


In [ ]:
df.head()

,rownames,minority,age,gender,credits,beauty,eval,division,native,tenure,students,allstudents,prof
0,1,yes,36,female,more,0.289916,4.3,upper,yes,yes,24,43,1
1,2,no,59,male,more,-0.737732,4.5,upper,yes,yes,17,20,2
2,3,no,51,male,more,-0.571984,3.7,upper,yes,yes,55,55,3
3,4,no,40,female,more,-0.677963,4.3,upper,yes,yes,40,46,4
4,5,no,31,female,more,1.509794,4.4,upper,yes,yes,42,48,5


In [ ]:
print(df.columns)

Index(['rownames', 'minority', 'age', 'gender', 'credits', 'beauty', 'eval',
       'division', 'native', 'tenure', 'students', 'allstudents', 'prof'],
      dtype='object')


In [ ]:
print("Mean:", df['eval'].mean())
print("Standard Deviation:", df['eval'].std())

Mean: 3.9982721382289412
Standard Deviation: 0.5548655656195949


# Probability of evaluation score > 4.5

In [ ]:
mean = df['eval'].mean()
std = df['eval'].std()
x = 4.5
z = (x - mean) / std
# probability P(X > 4.5)
probability = 1 - norm.cdf(z)
print("Mean =", mean)
print("Standard Deviation =", std)
print("Z-score =", z)
print("Probability of evaluation score > 4.5 =", probability)
print("Probability (%) =", probability * 100)

Mean = 3.9982721382289412
Standard Deviation = 0.5548655656195949
Z-score = 0.9042331924324778
Probability of evaluation score > 4.5 = 0.1829358816143234
Probability (%) = 18.29358816143234


**The probability of receiving an evaluation score greater than 4.5 is approximately 18.29%.**

# Probability of evaluation score > 3.5 and < 4.2
P(3.5 < X < 4.2)

In [ ]:
lower = 3.5
upper = 4.2
z_lower = (lower - mean) / std
z_upper = (upper - mean) / std
probability_range = norm.cdf(z_upper) - norm.cdf(z_lower)
print("Z-score lower =", z_lower)
print("Z-score upper =", z_upper)
print("Probability =", probability_range)
print("Probability (%) =", probability_range * 100)

Z-score lower = -0.8980051549469318
Z-score upper = 0.36356168821865525
Probability = 0.45731592936415527
Probability (%) = 45.731592936415524


**The probability of receiving an evaluation score between 3.5 and 4.2 is approximately 45%.**

# Two tailed z-test

In [ ]:
# Given values :
mu = 12
sigma = 5.5
sample_mean = 10.7
n = 36
alpha = 0.05

standard_error = sigma / np.sqrt(n)
z_stat = (sample_mean - mu) / standard_error
p_value = 2 * (1 - norm.cdf(abs(z_stat)))
print("Standard error =", standard_error)
print("Z-statistic:", z_stat)
print("P-value:", p_value)

Standard error = 0.9166666666666666
Z-statistic: -1.418181818181819
P-value: 0.15613768876786005


Null Hypothesis
$$ H_0: \mu = 12 $$

The mean score of the regional players is not significantly different from the historical mean of 12 points.

Alternative Hypothesis
$$ H_1: \mu \neq 12 $$

The mean score of the regional players is significantly different from the historical mean of 12 points.

In [ ]:
"critical_value = norm.ppf(1 - alpha / 2)
print("Critical values:", -critical_value, "and", critical_value)

if abs(z_stat) > critical_value:
    print("Reject the null hypothesis.")
else:
    print("Fail to reject the null hypothesis.")

Critical values: -1.959963984540054 and 1.959963984540054
Fail to reject the null hypothesis.


The calculated Z-statistic is -1.42, while the critical values at a 5% significance level for a two-tailed test are -1.96 and +1.96. Since the calculated Z-statistic lies within the acceptance region (-1.96, 1.96), we fail to reject the null hypothesis

The observed sample mean of 10.7 points is lower than 12, but the difference is not statistically significant at the 5% significance level.